src/debugger/nodeDebugWrapper.ts

```diff
@@ -250,4 +252,21 @@ export class NodeDebugWrapper {
     private isNullOrUndefined(value: any): boolean {
         return typeof value === "undefined" || value === null;
     }
+
+    /**
+     * Parses settings.json file for workspace root property
+     */
+    private getProjectRoot(args: any): string {
+        try {
+            let vsCodeRoot = path.resolve(args.program, "../..");
+            let settingsPath = path.resolve(vsCodeRoot, ".vscode/settings.json");

```

Comment:
You're right, I forgot that we have to specify the path in `launch.json`. Looking at https://code.visualstudio.com/docs/editor/tasks#_variable-substitution it seems that the variables which can be substituted are limited, but it does include environment variables, so we _could_ do something a bit hacky and have the extension set a `VSCODE_REACT_NATIVE_PROJECT_ROOT` environment variable and reference that, but I don't like that idea much.

Although, looking at the vscode codebase I do see a test with https://github.com/Microsoft/vscode/blob/ea26b957f6c593410724cd341e1db2c7c37170b8/src/vs/workbench/services/configurationResolver/test/node/configurationResolverService.test.ts#L117 which might suggest that we could reference `${config['react-native-tools'].projectRoot}` or something and it might work out?

If not, I think the current approach is ok